### DISCUSSION TITLE
Create connections without specifying node inputs/outputs

### DISCUSSION THREAD

**Discussion by StefanPopa99** [Foblex/f-flow#88] [Q&A] [answered]
Is it possible to create connections between nodes without directly specifying inputId and outputId? I find it way too complicated to always have to define the input and output id for (almost) each node. I understand that in this way it gives maximum flexibility, but for most of the time I don't need it. Instead, i'd want to make the connection between node ids.

Something like node1: {key: 'node1'} node2: {key: 'node2'} connection: {from: 'node1', to: 'node2'}.

**Comment by siarheihuzarevich** [comment, marked as answer]
Hi! Thank you for your question and for using Foblex Flow!

In the current version of Foblex Flow, connections between nodes are created using inputId and outputId, which provides maximum flexibility and allows for managing multiple ports. However, I understand that this can feel overly complex in most cases, especially when nodes have only one input and one output.

To simplify the process, you can use the node IDs directly as the input and output IDs. For example, set the inputId and outputId to be the same as the nodeId. This should make it much easier to create connections without the need to specify each port explicitly.

I hope this solution helps! If you have any additional questions or suggestions, feel free to reach out.

**Reply by StefanPopa99** [reply]
Thank you for the reply! I thought that the ids of inputs / outputs should be unique for internal processing. I'm completely fine to provide the same ids for both input and output. It is a bit of boilerplate, but i will have an utility function :)

**Comment by siarheihuzarevich** [comment]
✅ Feature added in version 17.6.0

Hi! This feature has been implemented and released in v17.6.0.
Feel free to upgrade and try it out. If you notice anything unexpected — let us know!

Thanks for your feedback and contribution 💪
